extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,141 @@
+        version=n.conanfile.version,
+        licenses=licenses(n.conanfile.license),
+        bom_ref=purl.to_string(),
+        purl=purl,
+        description=n.conanfile.description
+    )
+    if n.conanfile.homepage:
+        result.external_references.add(ExternalReference(
+            type=ExternalReferenceType.WEBSITE,
+            url=XsUri(n.conanfile.homepage),
+        ))
+    return result
+
+
+def format_cyclonedx_14_json(result):
+    from cyclonedx.output.json import JsonV1Dot4
+    serialized_json = JsonV1Dot4(result).output_as_string()
+    cli_out_write(serialized_json)
+
+
+def format_text(result):
+    out = ConanOutput()
+    out.error("Format 'text' not supported")
+
+
+@conan_command(group="Recipe", formatters={
+    "text": format_text,  # added by default in BaseConanCommand.__init__
+    "cyclonedx_1.4_json": format_cyclonedx_14_json})
+def create_sbom(conan_api: ConanAPI, parser, *args):
+    """
+    creates an SBOM in CycloneDX 1.4 JSON format
+    """
+    try:
+        import cyclonedx
+    except ModuleNotFoundError:
+        sys.stderr.write(
+            "The sbom extension needs an additional package, please run 'pip install cyclonedx-python-lib'\n")
+        sys.exit(1)
+    from cyclonedx.model.bom import Bom
+    # BEGIN COPY FROM conan: cli/commands/graph.py
+    common_graph_args(parser)
+    args = parser.parse_args(*args)
+    validate_common_graph_args(args)
+    cwd = os.getcwd()
+    path = conan_api.local.get_conanfile_path(args.path, cwd, py=None) if args.path else None
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile,
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update)
+    # END COPY
+    name.unknown_name_created = False
+    components = {n: create_component(n) for n in deps_graph.nodes}
+    bom = Bom()
+    bom.metadata.component = components[deps_graph.root]
```

Comment:
please add your own to `bom.metadata.tools`, so people know how the SBOM was built.
remmeber: `bom.metadata.tools` is prepoppulated, so just use `bom.metadata.tools.add(Tool(...))`

Reply:
Added